core/controllers/question_test.py

```diff
@@ -57,57 +63,256 @@ def setUp(self):
 
         # Update the acquired skill IDs for the exploration.
         collection_services.update_collection(
-            self.owner_id, self.collection_id, [{
+            self.new_user_id, self.collection_id, [{
                 'cmd': collection_domain.CMD_EDIT_COLLECTION_NODE_PROPERTY,
                 'property_name': (
                     collection_domain.COLLECTION_NODE_PROPERTY_ACQUIRED_SKILL_IDS), # pylint: disable=line-too-long
                 'exploration_id': self.exp_id,
                 'new_value': [self.skill_id]
             }], 'Update skill')
 
-        question = question_domain.Question(
+        self.question = question_domain.Question(
             'dummy', 'A Question',
             exp_domain.State.create_default_state('ABC').to_dict(), 1,
             self.collection_id, 'en')
 
-        question_id = question_services.add_question(self.owner_id, question)
+    def test_delete(self):
+        question_id = question_services.add_question(
+            self.new_user_id, self.question)
+        question_services.add_question_id_to_skill(
+            question_id, self.collection_id, self.skill_id,
+            self.new_user_id)
+        self.login(self.NEW_USER_EMAIL)
+        response = self.testapp.delete(
+            '%s/%s/%s' % (
+                feconf.QUESTION_DATA_URL, self.collection_id, question_id),
+            expect_errors=False)
+        self.assertEqual(response.status_int, 200)
+
+        response = self.testapp.delete(
+            '%s/%s' % (
+                feconf.QUESTION_DATA_URL, self.collection_id),
+            expect_errors=True)
+        self.assertEqual(response.status_int, 404)
+
+        response = self.testapp.delete(
+            '%s/' % feconf.QUESTION_DATA_URL, expect_errors=True)
+        self.assertEqual(response.status_int, 404)
+
+        self.logout()
+        self.login(self.random_email)
+        response = self.testapp.delete(
+            '%s/%s/%s' % (
+                feconf.QUESTION_DATA_URL, self.collection_id, question_id),
+            expect_errors=True)
+        self.assertEqual(response.status_int, 401)
+
+    def test_post(self):
+        payload = {}
+        payload['question'] = self.question.to_dict()
+        payload['skill_id'] = self.skill_id
+        self.login(self.NEW_USER_EMAIL)
+        response_json = self.post_json(
+            '%s' % feconf.QUESTION_CREATION_URL, payload,
+            expect_errors=False)
+        self.assertIn('question_id', response_json.keys())
+
+        del payload['skill_id']
+        self.post_json(
+            '%s' % feconf.QUESTION_CREATION_URL, {'payload': (
```

Comment:
Why do you still have a payload key here? This should follow exactly the way you did it in lines 115-117.

Ditto below.

Reply:
Updated.